### PR TITLE
JupyterLab 3 update

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,29 +10,43 @@ jobs:
...
+      uses: goanpeca/setup-miniconda@v1
       with:
-        python-version: '3.7'
-        architecture: 'x64'
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install jupyterlab
+        mamba-version: "*"
+        channels: conda-forge/label/jupyterlab_rc, conda-forge/label/jupyterlab_server_rc, conda-forge/label/jupyterlab_widgets_rc, conda-forge
```
Comment (jtpio) [inline .github/workflows/build.yml:20]: We should now be able to drop the prerelease labels:

```suggestion
        channels: conda-forge
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -10,29 +10,43 @@ jobs:
...
+
+    - name: Mamba install dependencies
+      shell: bash -l {0}
+      run: mamba install python=3.8 pip nodejs
+
+    - name: Mamba install JupyterLab 3
+      shell: bash -l {0}
+      run: mamba install jupyterlab=3 ipywidgets jupyterlab_widgets
```
Comment (jtpio) [inline .github/workflows/build.yml:28]: This should install `jupyterlab_widgets` automatically now:

```suggestion
      run: mamba install jupyterlab=3 ipywidgets=7.6
```

**Comment by jtpio** [comment]
Maybe we could also use `skip_if_exists` from `jupyter-packaging` here too? (same as in bqplot and other lab extensions)

**Comment by longouyang** [comment]
Hi all,
First off - thanks so much for your work on ipytree. It's a super useful extension -- I use it to quickly navigate around in structured data.

Is HEAD on the `jlab3` branch of `martinRenou/ipytree` intended to be fully working?

I tried installing jlab3 HEAD as a local extension but am getting "Error displaying widget: model not found" when I try to use it. Let me know if there's any way I can help with the PR

Best,
Long

**Comment by martinRenou** [comment]
Thanks for the kind words! And thanks for adding some activity here, that reminds me I need to finish this.

> Is HEAD on the jlab3 branch of martinRenou/ipytree intended to be fully working?

I am not sure. But this PR needs some changes before merging.

I'll try to find time to finish this.

**Pull request merged**